Code/GraphMol/Descriptors/AUTOCORR3D.cpp

```diff
@@ -0,0 +1,240 @@
+
+
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/MolTransforms/MolTransforms.h>
+
+#include "AUTOCORR3D.h"
+#include "MolData3Ddescriptors.h"
+
+#include "GraphMol/PartialCharges/GasteigerCharges.h"
+#include "GraphMol/PartialCharges/GasteigerParams.h"
+#include <Numerics/EigenSolvers/PowerEigenSolver.h>
+
+#include <Numerics/Matrix.h>
+#include <Numerics/SquareMatrix.h>
+#include <Numerics/SymmMatrix.h>
+#include <boost/foreach.hpp>
+#include <math.h>
+#include <Eigen/Dense>
+#include <Eigen/SVD>
+#include <iostream>
+#include <Eigen/Core>
+#include <Eigen/QR>
+
+
+
+using namespace Eigen;
+namespace RDKit {
+namespace Descriptors{
+
+namespace {
+
+MolData3Ddescriptors moldata3D;
+
+double* retreiveMat(MatrixXd matrix) {
+     double* arrayd = matrix.data();
+     return arrayd;
+  }
+
+double* retreiveVect(VectorXd matrix) {
+     double* arrayd = matrix.data();
+     return arrayd;
+
+  }
+
+VectorXd getEigenVect(std::vector<double> v){
+      double* varray_ptr = &v[0];
+      Map<VectorXd> V(varray_ptr,v.size());
+      return V;
+  }
+
+
+double* AppendDouble(double *w, double* Append, int length, int pos){
+    for (int i=pos;i<pos+length;i++) {
+          w[i]=Append[i-pos];
+        }
+
+    return w;
+}
+
+
+double* GetGeodesicMatrix(double* dist, int lag,int numAtoms){
+    int sizeArray=numAtoms*numAtoms;
+    double *Geodesic = new double[sizeArray];
+    for (int i=0; i<sizeArray;i++) {
+      if (dist[i]==lag) Geodesic[i]=1;
+      else  Geodesic[i]=0;
+    }
+
+    return Geodesic;
+}
+
+
+double* get3DautocorrelationDesc(double* dist3D, double* dist, int numAtoms, const ROMol& mol) {
+
+   Map<MatrixXd> dm(dist3D, numAtoms,numAtoms);
+
+   double *w= new double[80];
+
+   std::vector<double> wp= moldata3D.GetRelativePol(mol);
+
+   VectorXd Wp = getEigenVect(wp);
+
+   std::vector<double> wm= moldata3D.GetRelativeMW(mol);
+
+   VectorXd Wm = getEigenVect(wm);
+
+   std::vector<double> wi= moldata3D.GetRelativeIonPol(mol);
+
+   VectorXd Wi = getEigenVect(wi);
+
+   std::vector<double> wv= moldata3D.GetRelativeVdW(mol);
+
+   VectorXd Wv = getEigenVect(wv);
+
+   std::vector<double> we= moldata3D.GetRelativeENeg(mol);
+
+   VectorXd We = getEigenVect(we);
+
+   std::vector<double>  wu = moldata3D.GetUn(numAtoms);
+
+   VectorXd Wu = getEigenVect(wu);
+
+   std::vector<double>  ws =  moldata3D.GetIState(mol);
+
+   VectorXd Ws = getEigenVect(ws);
+
+   std::vector<double> wr= moldata3D.GetRelativeRcov(mol);
+
+   VectorXd Wr = getEigenVect(wr);
+
+   MatrixXd Bi;
+   MatrixXd tmp;
+   double TDBmat[8][10];
+   double dtmp;
+
+  for (int i=0;i<10;i++){
+    double * Bimat = GetGeodesicMatrix(dist, i+1, numAtoms);
+    Map<MatrixXd> Bi(Bimat, numAtoms,numAtoms);
+    MatrixXd RBi=Bi.cwiseProduct(dm);
+    double Bicount=(double) Bi.sum();
+
+    tmp=Wu.transpose() * RBi * Wu / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
```

Comment:
I'm not familiar enough with the descriptor to be sure, but this is not always a great solution for dealing with NaN. Are you sure that this is right?